src/plugins/plugin-generic-pool.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+var shimmer = require('shimmer');
+
+function wrapModuleRoot2(Pool, api) {
+  shimmer.wrap(Pool.Pool.prototype, 'acquire', function(original) {
+    return function(callback, priority) {
+      return original.call(this, callback ? api.wrap(callback) : callback,
```

Comment:
The API for generic-pool@2 specifies that the priority is optional but the callback is not.  However, if the callback is falsey, I thought it would be best to pass it directly to the original function call so that generic-pool can, itself, handle the fact that it is falsey.